src/api/spec/factories/bs_requests.rb

```diff
@@ -4,16 +4,19 @@
     state "new"
 
     before(:create) do |request|
-      request.bs_request_actions << create(:bs_request_action)
-
-      user = create(:confirmed_user)
-
-      request.creator   = user.login
-      request.commenter = user.login
+      unless request.creator
+        user = create(:confirmed_user)
 
+        request.creator   = user.login
+        request.commenter = user.login
+      end
+      unless request.bs_request_actions.any?
+        request.bs_request_actions << create(:bs_request_action)
+      end
       # Monkeypatch to avoid errors caused by permission checks made
       # in user and bs_request model
-      User.current = user
+      User.current = user unless request.creator
+      User.current = User.find_by(login: request.creator) if request.creator
```

Comment:
Isn't better to have it in only one line?
```ruby
User.current = request.creator ? User.find_by(login: request.creator) : user
```
Or if you want something more "readable"... (I don't like it that much)
```ruby
User.current = if request.creator
   user
else
  User.find_by(login: request.creator)
end
```